include/spirv-tools/optimizer.hpp

```diff
@@ -519,7 +519,8 @@ Optimizer::PassToken CreateDeadInsertElimPass();
 // interface are considered live and are not eliminated. This mode is needed
 // by GPU-Assisted validation instrumentation, where a change in the interface
 // is not allowed.
-Optimizer::PassToken CreateAggressiveDCEPass(bool preserve_interface = false);
+Optimizer::PassToken CreateAggressiveDCEPass();
+Optimizer::PassToken CreateAggressiveDCEPassPreserveInterface();
```

Comment:
I think I have changed my mind.  If you do this you will break those that depend on the interface with the bool.

Can you leave it as two function:

```
Optimizer::PassToken CreateAggressiveDCEPass();
Optimizer::PassToken CreateAggressiveDCEPass(bool preserve_interface);
```

This should not force anybody to have to change their code again.  Is that right?

Reply:
That would be preferable.  Two breaks are worse than one.

If we don't like the bool operand, we can *deprecate* the one with the bool operand, and add CreateAggressiveDCEPassPreserveInterface

Reply:
No two breaks aren't worse. 2 breaks is the answer to not having 0 breaks. The original ABI should remain, the in between one is a short term temporary headache.

I wasn't sure we could do a bool and non-bool, but if we can I'll try that.